### PR TITLE
Exception Handling Part 2

### PR DESCRIPTION
**Pull request by andyday**
this is the 2nd part of exception handling logic.  

it uses Content-Type based response encoding to encode HttpResponseExceptions to proper responses.

still adding javadoc comments, but all functionality is in place.

### CODE REVIEW

**Comment by andyday** [comment]
this is ready for review.

File: xrpc/src/main/java/com/nordstrom/xrpc/server/http/Recipes.java [hunk 1]
```diff
@@ -115,10 +128,12 @@ public static FullHttpResponse newResponse(
    * Instead, content type will be as specified by the parameter contentType and content length will
    * be the length of the parameter contentLength.
    */
+  // TODO (AD): these recipes have become a hodge podge.
+  // Need to rethink and refactor at some point.
```
Comment (jkinkead) [inline xrpc/src/main/java/com/nordstrom/xrpc/server/http/Recipes.java]: +100 to this. :)

Comment (andyday) [reply xrpc/src/main/java/com/nordstrom/xrpc/server/http/Recipes.java]: ?

Comment (jkinkead) [reply xrpc/src/main/java/com/nordstrom/xrpc/server/http/Recipes.java]: I agree strongly that this needs rethinking & organization!

Comment (pdex) [reply xrpc/src/main/java/com/nordstrom/xrpc/server/http/Recipes.java]: xio `Response.Builder` might be the way to go.

**Comment by andyday** [comment]
closing until I have the full encode/decode concept fully formed and written.  also exploring using netty pipeline for at least some of this...

**Pull request closed without merging**